assets/js/data/cart/selectors.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Retrieves add cart data from the state.
+ *
+ * @param {Object} state The current state.
+ * @return {Object} The data to return.
+ */
+export const getCartData = ( state ) => {
+	return state;
+};
```

Comment:
I would split out the various elements of the cart state into their own selectors (which you've started to a degree). That way you can set good defaults for if the cart hasn't been retrieved yet instead of consuming code having to set those defaults.  So for instance, if you follow the suggested state shape I made in the `reducer`, you would end up with something like:

- `getCart`: returns `state.cartEntity` (and you can return an object populated with defaults if it isn't set yet).
- `getCartErrors`: returns `state.errors`
- `getMeta`: returns `state.meta` (you could keep `getApplyingCoupon` and `getRemovingCoupon` selectors too).